### PR TITLE
Deprecate `Series.applymap`

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Part of https://github.com/rapidsai/cudf/issues/10169

### CODE REVIEW

File: python/cudf/cudf/core/series.py [hunk 1]
```diff
@@ -1011,7 +1011,10 @@ def map(self, arg, na_action=None) -> "Series":
             result.name = self.name
             result.index = self.index
         else:
-            result = self.applymap(arg)
+            # TODO: switch to `apply`
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
```
Comment (bdice) [inline python/cudf/cudf/core/series.py:1016]: Can this be more specific?
```suggestion
                warnings.simplefilter("ignore::FutureWarning")
```

File: python/cudf/cudf/core/series.py [hunk 2]
```diff
@@ -2241,6 +2244,11 @@ def applymap(self, udf, out_dtype=None):
         4    105
         dtype: int64
         """
+        warnings.warn(
+            "Series.applymap is deprecated and will be removed "
+            "in a future cuDF release. Use Series.apply instead",
```
Comment (bdice) [inline python/cudf/cudf/core/series.py:2249]:
```suggestion
            "in a future cuDF release. Use Series.apply instead.",
```

**Review by bdice — APPROVED** [review]
Looks good, with two minor suggestions.

**Comment by brandon-b-miller** [comment]
rerun tests

**Comment by brandon-b-miller** [comment]
rerun tests

**Comment by brandon-b-miller** [comment]
rerun tests

**Pull request merged**